src/hotspot/share/opto/subnode.cpp

```diff
@@ -185,6 +189,18 @@ Node *SubINode::Ideal(PhaseGVN *phase, bool can_reshape){
       return new AddINode(in1, phase->intcon(-i->get_con()));
   }
 
+  // Convert "c0 - (x + c1)" into "(c0 - c1) - x"
+  if (in2->Opcode() == Op_AddI
+      && phase->type(in1)->isa_int() != NULL
+      && phase->type(in1)->isa_int()->is_con()
```

Comment:
Line 194 and 195 can be expressed as `in1->Opcode() == Op_ConI`, same for the next 2 lines.